src/lib/autocomplete/autocomplete.ts

```diff
@@ -54,11 +57,18 @@ export class MdAutocomplete {
     }
   }
 
+  /** Panel should hide itself when the option list is empty. */
+  _setVisibility() {
+    Promise.resolve().then(() => this.showPanel = !!this.options.length);
```

Comment:
Is this Promise based so that we invoke a change detection or wait for something to stabilize?